src/commands.cpp

```diff
@@ -1974,60 +1988,178 @@ void CommandGetUA::procresult()
+                 attributename == "+sigPubk"  ||
+                 attributename == "+sigCu255" )
+            {
+                // if own user's attribute and it's a public key, check against derived pubKey
+                if (user->userhandle == client->me)
+                {
+                    if (attributename == "+puEd255")
+                    {
+                        if (memcmp(data, client->signkey->pubKey, EdDSA::PUBLIC_KEY_LENGTH))
+                        {
+                            LOG_warn << "Public key for Ed25519 doesn't match. Updating...";
+                            client->putua("+puEd255", (byte *) client->signkey->pubKey, EdDSA::PUBLIC_KEY_LENGTH);
+                        }
+                        else
+                        {
+                            LOG_info << "Public key for Ed25519 matches derived public key.";
+                        }
+                    }
+                    else if (attributename == "+puCu255")
+                    {
+                        if (memcmp(data, client->chatkey->pubKey, ECDH::PUBLIC_KEY_LENGTH))
+                        {
+                            LOG_warn << "Public key for x25519 doesn't match. Updating...";
+                            client->putua("+puCu255", (byte *) client->chatkey->pubKey, ECDH::PUBLIC_KEY_LENGTH);
+                        }
+                        else
+                        {
+                            LOG_info << "Public key for x25519 matches derived public key.";
+                        }
+                    }
+                }
+
+                client->app->getua_result(data, datalen);
+            }
             else
             {
-                // We need to shave off our 8 byte IV first.
-                string iv;
-                iv.assign(d, 0, 8);
-                string payload;
-                payload.assign(d, 8, l - 8);
-                d = payload;
-                if (!PaddedCBC::decrypt(&d, &client->key, &iv))
-                {
-                    delete[] data;
+                // build the TLV records
+                tlvRecords = TLVstore::containerToTLVrecords(string((char*) data, datalen));
+                if (!tlvRecords)
+                {
+                    LOG_err << "Cannot extract TLV records for public attribute " << attributename;
                     client->app->getua_result(API_EINTERNAL);
+                    delete [] data;
                     return;
                 }
+
+                client->app->getua_result(tlvRecords);
+                delete tlvRecords;
             }
-            client->app->getua_result((byte*)d.data(), d.size());
-        }
-        else
-        {
-            client->app->getua_result(data, l);
+            break;
+
+        case '#':   // protected
+            client->app->getua_result(data, datalen);
+            break;
+
+        default:    // legacy private attributes or unknown attribute
+            if (attributename == "firstname"    ||
```

Comment:
according to your notes on the wiki, `firstname` and `lastname` are protected:
https://wiki.developers.mega.co.nz/MegaHttpApi/UserAttributes